Events API v2 Support

jjinno opened the pull request.

jjinno committed:
Update docs for Events API v2

jjinno committed:
Create events_v2.py

Created Events class compatible with Events API v2 (not to be confused with REST API v2)
New API comes with new documented required fields and new optional fields.
Documentation links provided at top of code for reference to the PD-CEF & Events API v2

Implemented enqueue() to allow for payload to be specified by caller.
This allows direct use of the JSON produced for the request (better for unit-testing enqueue function)

Note that summary/source/severity are still required, and throw a KeyError if not found in either kwargs or payload.

jjinno committed:
Create event_request_v2.json

jjinno committed:
Add Events API v2 test

Similar to v1 test: directly tests the enqueue() function with a built-in event_action of "trigger"

jjinno committed:
Fixed flake8 bugs

jjinno committed:
moved images/links out of payload

https://v2.developer.pagerduty.com/docs/send-an-event-events-api-v2

Shows 'images' and 'links' outside of payload.

houqp commented:
@cugini-dbx any thing blocking this PR from getting merged?

cugini-dbx commented on pygerduty/events_v2.py:
```diff
@@ -0,0 +1,81 @@
+        # Required (according to documentation) PD-CEF fields
+        data = {"event_action": kwargs['event_action']}
+        data['payload'] = kwargs.get('payload', {})
+        for key in ['summary', 'source', 'severity']:
```
A teammate pointed out that `payload` is optional, but looks like this assumes payload is required (by checking for these three keys).

jjinno replied:
Thanks for pointing this out. 👍

cugini-dbx reviewed (commented):
Sorry for the delay in review. I just have one small comment around a some required field validation, otherwise it looks good to me!

jjinno committed:
Made payload optional

Even though payload is now optional, if it is specified, the fields within it are still checked as required

jjinno committed:
Fixed flake8 issues

jjinno committed:
fixed whitespace issue

jjinno commented:
The rest of the Travis issues appear to be not associated with this PR... not sure how best to resolve them

cugini-dbx commented:
_looks at watch_
... Well I totally dropped the ball, apologies for the very late return. The tests have all been fixed, so once you rebase, I'll be happy to merge this in.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/dropbox/pygerduty?pullRequest=84) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/dropbox/pygerduty?pullRequest=84) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/dropbox/pygerduty?pullRequest=84) it.</sub>

sdenef-adeo commented:
Hello,

I'm interested in this feature.
It seems all is OK to merge it AFAIU.

Regards,
Sébastien
